cache/contenthash/filehash.go

```diff
@@ -46,7 +47,7 @@ func NewFromStat(stat *fstypes.Stat) (hash.Hash, error) {
 	fi := &statInfo{stat}
 	hdr, err := tar.FileInfoHeader(fi, stat.Linkname)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "failed to stat file %s", stat.Path)
```

Comment:
"failed to checksum file" (or hash), as this error may not come from stat.

Reply:
Makes sense, thanks! I've updated it, PTAL.